File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        res = 0
        tot = sum(apple)
        capacity = sorted(capacity, reverse=True)
        for c in capacity:
            res += 1
            if tot - c <= 0:
                break
            tot -= c
        return res

File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.cpp
class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& capacity) {
        int res = 0;
        int tot = accumulate(apple.begin(), apple.end(), 0);
        sort(capacity.begin(), capacity.end(), greater<int>());
        for(int i = 0; i < capacity.size(); i++)
        {
            res++;
            if(tot - capacity[i] <= 0)
            {
                break;
            }
            tot -= capacity[i];
        }
        return res;
    }
};